File: Easy/Leetcode-989-Add-to-Array-Form/AddToArrayForm.java
import java.util.*;

public class AddToArrayForm {
  public static List<Integer> addToArrayForm(int[] num, int k) {
    int i = num.length - 1;
    int carry = k;
    List<Integer> ans = new ArrayList<>();

    while(i>=0 || carry>0) {
      if(i>=0) {
        int res = num[i] + carry;
        ans.add(0, res % 10);
        carry = res / 10;
      } else {
        ans.add(0, carry % 10);
        carry /= 10;
      }
      i--;
    }
    return ans;
  }
  public static void solve() {
    int[] num = {9, 9, 9, 9};
    int k = 1;
    List<Integer> ans = addToArrayForm(num, k);
    System.out.println("--------------"+ ans);
  }
  public static void main(String[] args) {
    solve();
  }
}

File: Easy/Leetcode-989-Add-to-Array-Form/AddToArrayForm.js
var addToArrayForm = function (num, k) {
  var i = num.length - 1;
  var carry = k;
  var res = [];

  while (i >= 0 || carry > 0) {
    if (i >= 0) {
      var rem = num[i] + carry;
      res.push(rem % 10);
      carry = Math.trunc(rem / 10);
    } else {
      res.push(carry % 10);
      carry = Math.trunc(carry / 10);
    }
    i--;
  }
  return res.reverse();
};

var num = [1, 2, 0, 0];
var k = 34;
var ans = addToArrayForm(num, k);
console.log("-----------", ans);
